### PR TITLE
gzdoom: update to version 4.10.0.

### PR DESCRIPTION
**Pull request by SpidFightFR**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x64-GlibC)

#### Additional notes for the PR:

- Remade the i686 patch as it was broken after updating.
- Removed the sdl patch as it's already applied in the gzdoom repo ( see [sdlglvideo.cpp](https://github.com/ZDoom/gzdoom/blob/c675d60bacefd9c6651b8006f799f7bb55d0870b/src/common/platform/posix/sdl/sdlglvideo.cpp#L246) )
- Removed the vsed lines post-patch as gitinfo.cpp has been removed since [release 4.9.0](https://github.com/ZDoom/gzdoom/releases/tag/g4.9.0), also removed the leftover variable "tagdate".
- Added `libvpx-devel` as a make dependency as it doesn't build otherwise.

### CODE REVIEW

**Comment by SpidFightFR** [comment]
So the i686 patch doesn't seem to work. If anyone has a suggestion, feel free to bring one !

**Comment by classabbyamp** [comment]
```patch
--- a/srcpkgs/gzdoom/template
+++ b/srcpkgs/gzdoom/template
@@ -21,6 +21,7 @@ nocross=yes
 CXXFLAGS=-std=gnu++11
 
 if [ "$XBPS_TARGET_LIBC" = musl ]; then
+       CFLAGS="-D_7ZIP_AFFINITY_DISABLE"
        makedepends+=" musl-fts-devel libexecinfo-devel"
 fi
 ```
should fix musl

File: srcpkgs/gzdoom/patches/fix-i686.patch [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
- 			APPEND_STRING PROPERTY COMPILE_FLAGS " -msse2 -mmmx" )
- 	endif()
+-	set_property( SOURCE
++        set_source_files_properties(
++                common/rendering/polyrenderer/poly_all.cpp
++		rendering/swrenderer/r_all.cpp
+ 		common/utility/palette.cpp
+ 		common/utility/x86.cpp
+-		rendering/swrenderer/r_all.cpp
+ 		APPEND_STRING PROPERTY COMPILE_FLAGS " ${SSE2_ENABLE}" )
```
Comment (classabbyamp) [inline srcpkgs/gzdoom/patches/fix-i686.patch:16]:
```suggestion
-		APPEND_STRING PROPERTY COMPILE_FLAGS " ${SSE2_ENABLE}" )
+ 		APPEND_STRING PROPERTY COMPILE_FLAGS " -msse2 -mfpmath=sse" )
```

**Review by classabbyamp — COMMENTED** [review]
```patch
-case "$XBPS_TARGET_MACHINE" in
-	i686*) configure_args+=" -DSSE_MATTERS=true" ;;
-esac
-
```

this should fix the current error on i686, but it seems that they no longer want to support i686:
```
../src/common/engine/i_interface.cpp:8:29: error: static assertion failed: 32 builds are not supported
```

File: srcpkgs/gzdoom/patches/fix-i686.patch (hunk 1 above)
Comment (SpidFightFR) [reply srcpkgs/gzdoom/patches/fix-i686.patch]: Alright, i make a little test with what you sent, thanks again !

If everything works out, i'm gonna re-edit the template to remove the comments and re-commit. It's currently just a test.

Comment (SpidFightFR) [reply srcpkgs/gzdoom/patches/fix-i686.patch]: alright i'm aware some things went wrong here, but i'm doing another test, this one it should be good.

Comment (SpidFightFR) [reply srcpkgs/gzdoom/patches/fix-i686.patch]: damn... i really hoped it would work... :(

File: srcpkgs/gzdoom/template [hunk 2]
```diff
@@ -25,10 +24,6 @@ if [ "$XBPS_TARGET_LIBC" = musl ]; then
 	makedepends+=" musl-fts-devel libexecinfo-devel"
```
Comment (classabbyamp) [inline srcpkgs/gzdoom/template:24]: add `CFLAGS="-D_7ZIP_AFFINITY_DISABLE"` here

File: srcpkgs/gzdoom/template [hunk 3]
```diff
@@ -1,21 +1,20 @@
 # Template file for 'gzdoom'
 pkgname=gzdoom
-version=4.7.1
-revision=2
-_tagdate=2021-10-20
+version=4.10.0
+revision=1
```
Comment (classabbyamp) [inline srcpkgs/gzdoom/template:4]: add `archs="~i686* ~arm*"`

**Review by classabbyamp — CHANGES_REQUESTED** [review]

**Pull request merged**